src/core/lib/iomgr/tcp_client_posix.cc

```diff
@@ -224,10 +228,12 @@ static void on_writable(void* acp, grpc_error* error) {
     gpr_free(error_descr);
     gpr_free(desc);
     error = grpc_error_set_str(error, GRPC_ERROR_STR_TARGET_ADDRESS,
-                               grpc_slice_from_copied_string(ac->addr_str));
+                               addr_str_slice);
+    grpc_slice_unref(addr_str_slice);
```

Comment:
This shouldn't be needed I think. grpc_error_set_str takes ownership of the slice and it will be unref-ed in the error destructor

Reply:
You are right, but how would I know `grpc_error_set_str` takes ownership? There's no documentation. Still, I'd to keep the unref, makes "sense" as a pattern, and it's more robust if the way the slice is used is later changed to something that doesn't take ownership.

Reply:
But isn't this changing the behavior from what it was before? Before, the call was:
```
error = grpc_error_set_str(error, GRPC_ERROR_STR_TARGET_ADDRESS,
                               grpc_slice_from_copied_string(ac->addr_str));
```

grpc_slice_from_copied_string creates a slice with a ref of one, and that one ref is "handed off" to the error.

Now there is an additional unref that wasn't there before. I am actually surprised that the tests didn't fail for one of those two behaviors... 

I agree that documentation is needed for `grpc_error_set_str`